daml-lf/transaction/src/main/scala/com/digitalasset/daml/lf/transaction/TransactionCoder.scala

```diff
@@ -274,10 +278,19 @@ object TransactionCoder {
           children <- childrenOrError
           cv <- decodeVal(protoExe.getChosenValue)
           templateId <- ValueCoder.decodeIdentifier(protoExe.getTemplateId)
-          controllers <- toPartySet(protoExe.getControllersList)
+          actingParties <- toPartySet(protoExe.getActorsList)
+          encodedControllers <- toPartySet(protoExe.getControllersList)
+          controllers <- if (!(txVersion precedes minNoControllers)) {
+            if (encodedControllers.isEmpty) {
+              Right(actingParties)
+            } else {
+              Left(DecodeError(s"$txVersion is to new to have exercise controllers"))
```

Comment:
```suggestion
              Left(DecodeError(s"$txVersion is too new to have exercise controllers"))
```

Reply:
I would also phrase it more like: `As of version $txVersion, exercise controllers must be empty`.

Reply:
done